From: KevinDuan <duankaiwen@kylinos.cn>
Date: Tue, 23 Jul 2024 13:05:04 +0000
Subject: Translated using Weblate (Kyrgyz)

Currently translated at 100.0% (155 of 155 strings)

Translation: openkylin-nile-new/kylin-vpn
Translate-URL: http://weblate.openkylin.top/projects/openkylin-nile-new/kylin-vpn/ky/
---
 src-vpn/translations/kylin-vpn_kk.ts | 302 +++++++++++++++++------------------
 src-vpn/translations/kylin-vpn_ky.ts | 302 +++++++++++++++++------------------
 2 files changed, 302 insertions(+), 302 deletions(-)

diff --git a/src-vpn/translations/kylin-vpn_kk.ts b/src-vpn/translations/kylin-vpn_kk.ts
index 2e8dd36..bccd30a 100644
--- a/src-vpn/translations/kylin-vpn_kk.ts
+++ b/src-vpn/translations/kylin-vpn_kk.ts
@@ -6,17 +6,17 @@
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="79"/>
         <source>Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>تەڭشەۋلەر</translation>
     </message>
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="141"/>
         <source>Kylin VPN</source>
-        <translation type="unfinished"></translation>
+        <translation>اسباپبٸVPN</translation>
     </message>
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="144"/>
         <source>kylin vpn applet desktop message</source>
-        <translation type="unfinished"></translation>
+        <translation>VPNتەڭگەرگٸشتٸ ۇستەل بەتىندە كورسەتۋ</translation>
     </message>
 </context>
 <context>
@@ -24,391 +24,391 @@
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="827"/>
         <source>MPPE encryption algorithm:</source>
-        <translation type="unfinished"></translation>
+        <translation>MPPE سيفىرلاستىرۋ ەسەپتەۋگە ٴتاسٸلٸ:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="828"/>
         <source>Use Stateful encryption</source>
-        <translation type="unfinished"></translation>
+        <translation>ٸستەتۋ كۇيى سيفرلاسقان</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="829"/>
         <source>Send PPP echo packets</source>
-        <translation type="unfinished"></translation>
+        <translation>pppقايتپ كورسەتۋ سومكاسىنا جولداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="830"/>
         <source>Authentication Mode:</source>
-        <translation type="unfinished"></translation>
+        <translation>دالەلدەۋ پىشىنى:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="831"/>
         <source>PAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>PAP دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="832"/>
         <source>CHAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>CHAP دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="833"/>
         <source>MSCHAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>MSCHAP دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="834"/>
         <source>MSCHAP2 authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>MSCHAP2 دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="835"/>
         <source>EAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>EAP دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="836"/>
         <source>Compression Mode:</source>
-        <translation type="unfinished"></translation>
+        <translation>پېرىسلاش كۇيى:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="837"/>
         <source>Allow BSD data compression</source>
-        <translation type="unfinished"></translation>
+        <translation>BSD نى پېرىسلاشقا جول قويۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="838"/>
         <source>Allow Default data compression</source>
-        <translation type="unfinished"></translation>
+        <translation>Defaultنى پېرىسلاشقا جول قويۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="839"/>
         <source>Allow TCP header compression</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP نى پېرىسلاشقا جول قويۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="840"/>
         <source>Use protocol field compression negotiation</source>
-        <translation type="unfinished"></translation>
+        <translation>ٸستەتۋ كەلىسىمى ساباعىنداعى پېرىسلاش كەلىسىمى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="841"/>
         <source>Use Address/Control compression</source>
-        <translation type="unfinished"></translation>
+        <translation>ٵدىرىس/كونتىروللاپ پېرىسلاشنى ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="847"/>
         <source>All Available</source>
-        <translation type="unfinished"></translation>
+        <translation>بارلٸق بارلار</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="848"/>
         <source>128-bit</source>
-        <translation type="unfinished"></translation>
+        <translation>128 بەتالىق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="849"/>
         <source>40-bit</source>
-        <translation type="unfinished"></translation>
+        <translation>40 بەتالىق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1046"/>
         <source>Use custom gateway port</source>
-        <translation type="unfinished"></translation>
+        <translation>انٸقتاما بەرۋ تور وتكەلى اۋزىن ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1047"/>
         <source>Use compression</source>
-        <translation type="unfinished"></translation>
+        <translation>پېرىسلاپ ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1048"/>
         <source>Use a TCP connection</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP اۋلاۋ تاسٸلدٸ قولدانۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1049"/>
         <source>Set virtual device type</source>
-        <translation type="unfinished"></translation>
+        <translation>جاساندى اسباب تيبىن بەلگٸلەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1050"/>
         <source>Set virtual device name</source>
-        <translation type="unfinished"></translation>
+        <translation>جاساندى اسباب اتاعىن بەلگٸلەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1051"/>
         <source>Limit TCP Maximum Segment Size(MSS)</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP ەڭ ۇلكەن بولٸك ۇلكەنلۇقىنى شەكتەۋ(MSS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1052"/>
         <source>Randomize remote hosts</source>
-        <translation type="unfinished"></translation>
+        <translation>ئىختىيارىيلاشقان كۇيدە باس اپپاراتىن جىراقتان كونتىرول ەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1053"/>
         <source>IPv6 tun link</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv6 tunاۋلاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1054"/>
         <source>Specify ping interval</source>
-        <translation type="unfinished"></translation>
+        <translation>pingدەۋرىيلىكىنى بەلگٸلەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1055"/>
         <source>Specify exit or restart ping</source>
-        <translation type="unfinished"></translation>
+        <translation>شەگىنۋ ياكي قاتە قوزعالعان ping نى كورسەتىپ بەرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1056"/>
         <source>Specify max routes</source>
-        <translation type="unfinished"></translation>
+        <translation>شەكتى لەنيانى كورسەتىپ بەرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1057"/>
         <source>Infinite retry on error</source>
-        <translation type="unfinished"></translation>
+        <translation>قاتەلىك شىققاندا باساسٸز كۇيدە قاتە قايتالاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1058"/>
         <source>Use custom key size</source>
-        <translation type="unfinished"></translation>
+        <translation>انٸقتاما بەرٸلگەن قۇپيا كىلت ۇلكەن كىشكەنەلىگىن ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1059"/>
         <source>Choose</source>
-        <translation type="unfinished"></translation>
+        <translation>تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1062"/>
         <source>Use custom renegotiation interval</source>
-        <translation type="unfinished"></translation>
+        <translation>ٶز الدىنا قاتە سۇحباتتاسۋ ارالىعىن ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1064"/>
         <source>Use custom tunnel Maximum Transmission Umit(MTU)</source>
-        <translation type="unfinished"></translation>
+        <translation>ەرەكشە تۇنەلدىڭ ەڭ ۇلكەن جولدانبا ۇزاتۋ مۇئەسسەسەسى(MTU) نى ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1066"/>
         <source>Use custom UDP fragment size</source>
-        <translation type="unfinished"></translation>
+        <translation>انٸقتاما بەرٸلگەن UDP پارشالاۋ ۇلكەندىگىن ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1068"/>
         <source>Accept authenticated packets from any address (Float)</source>
-        <translation type="unfinished"></translation>
+        <translation>ٴارقانداي ٵدىرىس قابىلداۋ ەتۋ ( Float ) ازاماتتىق تەكسەرۋ ارقىلى ساندىق مالىمەت بوقشاسٸ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1070"/>
         <source>Subject Match</source>
-        <translation type="unfinished"></translation>
+        <translation>باس تاقٸرٸبٸن سايكەستىرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1071"/>
         <source>Key File</source>
-        <translation type="unfinished"></translation>
+        <translation>شەشۋشٸ حۇجات</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1072"/>
         <source>Key Direction</source>
-        <translation type="unfinished"></translation>
+        <translation>شەشۋشٸ باعىت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1074"/>
         <source>Server Address</source>
-        <translation type="unfinished"></translation>
+        <translation>قىزىمەت وتەۋ اسبابى ٵدىرىسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1075"/>
         <source>Port</source>
-        <translation type="unfinished"></translation>
+        <translation>اۋٸزٸ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1076"/>
         <source>Proxy USername</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكىلەتىنەن ابونىت مى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1077"/>
         <source>Proxy Password</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكىلەتىنەن قۇپيا نومەر</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1079"/>
         <source>General</source>
-        <translation type="unfinished"></translation>
+        <translation>ادەتتەگٸ تارتىپ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1080"/>
         <source>TLS settings</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS تەڭشەۋلەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1081"/>
         <source>Server Certificate Check</source>
-        <translation type="unfinished"></translation>
+        <translation>قىزىمەت وتەۋ اسبابى كنەشكاسن تەكسەرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1083"/>
         <source>Use the previous authentication end (server) certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>بۇرٸنعٸ دالەلدەۋ سوڭىن (قىزىمەت وتەۋ اسبابى) كنەشكاسن ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1085"/>
         <source>Verify peer (server) certificate nsCertType specification</source>
-        <translation type="unfinished"></translation>
+        <translation>تەڭ نۇكتە (قىزىمەت وتەۋ اسبابى) كناشكاسى nsCertType نىڭ كونكرېتلىقىنى دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1086"/>
         <source>Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>فورٸم</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1087"/>
         <source>Proxies</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكىلەتىنەن</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1088"/>
         <source>Proxy Type</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكىلەتىنەن تۇرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1089"/>
         <source>Security</source>
-        <translation type="unfinished"></translation>
+        <translation>حاۋىپسىزدىك</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1090"/>
         <source>HMAC Authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>HMAC دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1100"/>
         <source>Input content</source>
-        <translation type="unfinished"></translation>
+        <translation>مازمۇن كىرگىزۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1102"/>
         <source>No</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1106"/>
         <source>Self-adaption</source>
-        <translation type="unfinished"></translation>
+        <translation>ٶز-ٶزىگە سايكەسۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1107"/>
         <source>Automatic</source>
-        <translation type="unfinished"></translation>
+        <translation>اۆتوماتىك</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1112"/>
         <source>Exit</source>
-        <translation type="unfinished"></translation>
+        <translation>شەگىنۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1113"/>
         <source>Restart</source>
-        <translation type="unfinished"></translation>
+        <translation>قاتە قوزعالتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1115"/>
         <source>Don&apos;t verify certificate identification</source>
-        <translation type="unfinished"></translation>
+        <translation>كناشكا ازاماتتىق دالەلدەۋدى ئىسپاتلىماڭ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1116"/>
         <source>Verify the entire subject exactly</source>
-        <translation type="unfinished"></translation>
+        <translation>پۇتٸن تاقٸرٸبٸن شٸنٸندا انىقتاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1117"/>
         <source>Verify name exactly</source>
-        <translation type="unfinished"></translation>
+        <translation>اتاعىن انىقتاپ دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1118"/>
         <source>Verify name by prefix</source>
-        <translation type="unfinished"></translation>
+        <translation>اتاعىن الدى قوسىلما بويىنشا دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1120"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1123"/>
         <source>Server</source>
-        <translation type="unfinished"></translation>
+        <translation>قىزىمەت وتەۋ اسبابى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1121"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1124"/>
         <source>Client</source>
-        <translation type="unfinished"></translation>
+        <translation>جيۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1126"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1130"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1139"/>
         <source>None</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1127"/>
         <source>TLS-Certification</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS-دالەلدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1128"/>
         <source>TLS-Encryption</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS-سيفىرلاستىرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1134"/>
         <source>Not Required</source>
-        <translation type="unfinished"></translation>
+        <translation>تالاپ ورىندالمايدى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1138"/>
         <source>Default</source>
-        <translation type="unfinished"></translation>
+        <translation>وز قالپٸ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1216"/>
         <source>Options:</source>
-        <translation type="unfinished"></translation>
+        <translation>تالدانبالار:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1219"/>
         <source>Request an inner IP address</source>
-        <translation type="unfinished"></translation>
+        <translation>ٸشكٸ IP ٵدىرىسى جابلماس ەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1220"/>
         <source>Enforce UDP encapsulation</source>
-        <translation type="unfinished"></translation>
+        <translation>UDP قاپتاۋدى اتقار ەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1221"/>
         <source>Use IP compression</source>
-        <translation type="unfinished"></translation>
+        <translation>IP  پېرىسلاش ٸستەتۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1222"/>
         <source>Enable custom password suggestions</source>
-        <translation type="unfinished"></translation>
+        <translation>ٶزى بەلگىلەەن قۇپيا نومەر قوزعالتۋ ۇسنستى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1382"/>
         <source>Choose a private key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكە كىلت تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1384"/>
         <source>Key Files (*.key *.pem *.der *.p12 *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>شەشۋشٸ حۇجاتتار (*.key *.pem *.der *.p12 *.pfx)</translation>
     </message>
 </context>
 <context>
@@ -416,96 +416,96 @@
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="331"/>
         <source>Type</source>
-        <translation type="unfinished"></translation>
+        <translation>تۇرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="332"/>
         <source>Name</source>
-        <translation type="unfinished"></translation>
+        <translation>مى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="333"/>
         <source>Static Key</source>
-        <translation type="unfinished"></translation>
+        <translation>تٸنٸش كۇيدەگى قۇپيا كىلت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="334"/>
         <source>Local IP</source>
-        <translation type="unfinished"></translation>
+        <translation>جەرلىك  ٵدىرىسIP</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="335"/>
         <source>Remote IP</source>
-        <translation type="unfinished"></translation>
+        <translation>جىراقتان كونتىرول ىستەيتىن ٵدىرىسIP</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="336"/>
         <source>PIN Code</source>
-        <translation type="unfinished"></translation>
+        <translation>PIN بەلگىسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="337"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="849"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="856"/>
         <source>Password</source>
-        <translation type="unfinished"></translation>
+        <translation>قۇپيا نومەر</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="338"/>
         <source>NT Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>NT كولەم</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="340"/>
         <source>Server Address</source>
-        <translation type="unfinished"></translation>
+        <translation>قىزىمەت وتەۋ اسبابى ٵدىرىسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="341"/>
         <source>Authentication Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>دالەلدەۋ فورماسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="342"/>
         <source>CA Certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>CA كىناشكاسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="343"/>
         <source>User Certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>پايدالانۋشٸ كناشكاسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="344"/>
         <source>Key Direction</source>
-        <translation type="unfinished"></translation>
+        <translation>شەشۋشٸ باعىت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="345"/>
         <source>Private Key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكە كىلت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="346"/>
         <source>Private Key Password</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكە كىلت پارولى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="347"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="349"/>
         <source>Password Options</source>
-        <translation type="unfinished"></translation>
+        <translation>پارول تالدانبالارٸ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="348"/>
         <source>Username</source>
-        <translation type="unfinished"></translation>
+        <translation>پايدالانۋشٸ مى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="352"/>
         <source>Notice:
 If key direction is used, it must be opposite to the VPN side used. If &apos;1&apos; is used, the connection must use &apos;0&apos;. If you are not sure which value to use, please contact your system administrator.</source>
-        <translation type="unfinished"></translation>
+        <translation>ديقات:قيسٸق قۇپيا كىلت باعىتتالۋى ٸستەتكەن بولسا، ول ٴسوزسٸز ٸستەتكەن VPN ولچى مەنەن قاراما﹣قارسى بولۋٸ كەرەك. قيسٸق «1» نى ٸستەتكەن بولسا، وللاشتا ٴسوزسٸز «0» نى ٸستەتۋ كەرەك. قيسٸق قايسى مانىن ٸستەتۋنى كەسٸم جاسايسٸزبا سەستيما باسقارۋشٸسٸ مەنەن حابارلاسٸڭٸز.</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="359"/>
@@ -513,104 +513,104 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="361"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="362"/>
         <source>Choose</source>
-        <translation type="unfinished"></translation>
+        <translation>تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="369"/>
         <source>None</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="373"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="379"/>
         <source>Save password only for this user</source>
-        <translation type="unfinished"></translation>
+        <translation>تەك كەزەكتەگى ابونىت قۇپيا نومەر ساقتاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="374"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="380"/>
         <source>Save password for all users</source>
-        <translation type="unfinished"></translation>
+        <translation>بارلٸق الارماندار ٷشٸن قۇپيا نومەر ساقتاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="375"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="381"/>
         <source>Ask password every time</source>
-        <translation type="unfinished"></translation>
+        <translation>ٵر رەت قۇپيا نومەر سۇراۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="376"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="382"/>
         <source>Don&apos;t require a password</source>
-        <translation type="unfinished"></translation>
+        <translation>قۇپيا نومەر تالاپ ورىنداماستىق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="385"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="386"/>
         <source>Required</source>
-        <translation type="unfinished"></translation>
+        <translation>ٴسوزسىز تولدٸرعٸش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="854"/>
         <source>Certificate(TLS)</source>
-        <translation type="unfinished"></translation>
+        <translation>كىناشكا(TLS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="855"/>
         <source>Static key</source>
-        <translation type="unfinished"></translation>
+        <translation>تٸنٸش كۇيدەگى كىلت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="857"/>
         <source>Password and certificate(TLS)</source>
-        <translation type="unfinished"></translation>
+        <translation>قۇپيا نومەر ۋا كناشكا(TLS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="866"/>
         <source>Certificate/Private key</source>
-        <translation type="unfinished"></translation>
+        <translation>كناشكا/قاسيەت كىلت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="867"/>
         <source>Certificate/ssh-agent</source>
-        <translation type="unfinished"></translation>
+        <translation>كىناشكا/ ssh-agent</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="868"/>
         <source>Smart card</source>
-        <translation type="unfinished"></translation>
+        <translation>وي قابٸلەتتٸ قاريتا</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="923"/>
         <source>Choose a private key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكە كىلت تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="925"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="964"/>
         <source>Key Files (*.key *.pem *.der *.p12 *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>شەشۋشٸ حۇجاتتار (*.key *.pem *.der *.p12 *.pfx)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="936"/>
         <source>Choose a CA certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>CA كناشكاسى تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="938"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="951"/>
         <source>CA Files (*.pem *.der *.p12 *.crt *.cer *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>CA حۇجاتتارى (*.pem *.der *.p12 *.crt *.cer *.pfx)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="949"/>
         <source>Choose a User certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>پايدالانۋشٸ كناشكاسى تالداۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="962"/>
         <source>Choose a Static key</source>
-        <translation type="unfinished"></translation>
+        <translation>تٸنٸش كۇيدەگى كىلت تالداۋ</translation>
     </message>
 </context>
 <context>
@@ -635,22 +635,22 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="172"/>
         <source>Advanced</source>
-        <translation type="unfinished"></translation>
+        <translation>جوعارعى دارەجە بۇلۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="177"/>
         <source>Auto Connection</source>
-        <translation type="unfinished"></translation>
+        <translation>اۆتوماتتى اۋلاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="178"/>
         <source>Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>كۇشىنەن قالدىرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="179"/>
         <source>Confirm</source>
-        <translation type="unfinished"></translation>
+        <translation>كەسٸم جاساۋ</translation>
     </message>
 </context>
 <context>
@@ -658,47 +658,47 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="170"/>
         <source>IPv4 Config</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv4 تەڭگەرگٸش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="171"/>
         <source>Address</source>
-        <translation type="unfinished"></translation>
+        <translation>ٵدىرىس</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="172"/>
         <source>Netmask</source>
-        <translation type="unfinished"></translation>
+        <translation>تارماق تور جاسىرىن نومەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="173"/>
         <source>Default Gateway</source>
-        <translation type="unfinished"></translation>
+        <translation>وز قالپٸ تور وتكەلى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="174"/>
         <source>DNS Server</source>
-        <translation type="unfinished"></translation>
+        <translation>DNS قىزىمەت وتەۋ تۇرلەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="175"/>
         <source>Search Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>تور اتاعىن ٸزدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="176"/>
         <source>DHCP Client ID</source>
-        <translation type="unfinished"></translation>
+        <translation>DHCP جيۋ IDسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="178"/>
         <source>Auto(DHCP)</source>
-        <translation type="unfinished"></translation>
+        <translation>اۆتوماتتى(DHCP)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="179"/>
         <source>Manual</source>
-        <translation type="unfinished"></translation>
+        <translation>قۇل ارقىلى ارەكەتتەندىرۋ</translation>
     </message>
 </context>
 <context>
@@ -706,42 +706,42 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="159"/>
         <source>IPv6 Config</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv6 تەڭگەرگٸش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="160"/>
         <source>Address</source>
-        <translation type="unfinished"></translation>
+        <translation>ٵدىرىس</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="161"/>
         <source>Netmask</source>
-        <translation type="unfinished"></translation>
+        <translation>تارماق تور جاسىرىن نومەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="162"/>
         <source>Default Gateway</source>
-        <translation type="unfinished"></translation>
+        <translation>وز قالپٸ تور وتكەلى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="163"/>
         <source>DNS Server</source>
-        <translation type="unfinished"></translation>
+        <translation>DNS قىزىمەت وتەۋ تۇرلەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="164"/>
         <source>Search Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>تور اتاعىن ٸزدەۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="166"/>
         <source>Auto(DHCP)</source>
-        <translation type="unfinished"></translation>
+        <translation>اۆتوماتتى(DHCP)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="167"/>
         <source>Manual</source>
-        <translation type="unfinished"></translation>
+        <translation>قۇل ارقىلى ارەكەتتەندىرۋ</translation>
     </message>
 </context>
 <context>
@@ -749,19 +749,19 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="61"/>
         <source>Not connected</source>
-        <translation type="unfinished"></translation>
+        <translation>جالعانباعان</translation>
     </message>
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="146"/>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="161"/>
         <source>Disconnect</source>
-        <translation type="unfinished"></translation>
+        <translation>ٷزٸلۋ</translation>
     </message>
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="148"/>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="159"/>
         <source>Connect</source>
-        <translation type="unfinished"></translation>
+        <translation>جالعانۋ</translation>
     </message>
 </context>
 <context>
@@ -774,7 +774,7 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/single-pages/vpnpage.cpp" line="264"/>
         <source>VPN Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN تەڭشەۋلەرى</translation>
     </message>
 </context>
 <context>
@@ -782,43 +782,43 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="58"/>
         <source>create VPN</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN قۇرۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="92"/>
         <source>VPN Type</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN تۇرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="120"/>
         <source>VPN Name</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN مى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="126"/>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="147"/>
         <source>Required</source>
-        <translation type="unfinished"></translation>
+        <translation>ٴسوزسىز تولدٸرعٸش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="142"/>
         <source>VPN Server</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN قىزىمەت وتەۋ اسبابى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="168"/>
         <source>Auto Connection</source>
-        <translation type="unfinished"></translation>
+        <translation>اۆتوماتتى اۋلاۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="170"/>
         <source>Confirm</source>
-        <translation type="unfinished"></translation>
+        <translation>كەسٸم جاساۋ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="172"/>
         <source>Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>كۇشىنەن قالدىرۋ</translation>
     </message>
 </context>
 <context>
@@ -833,7 +833,7 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpnobject.cpp" line="50"/>
         <source>vpn tool</source>
-        <translation type="unfinished"></translation>
+        <translation>vpn قۇرالى</translation>
     </message>
 </context>
 </TS>
diff --git a/src-vpn/translations/kylin-vpn_ky.ts b/src-vpn/translations/kylin-vpn_ky.ts
index ecae730..078125d 100644
--- a/src-vpn/translations/kylin-vpn_ky.ts
+++ b/src-vpn/translations/kylin-vpn_ky.ts
@@ -6,17 +6,17 @@
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="79"/>
         <source>Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>تەڭشەكتەر</translation>
     </message>
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="141"/>
         <source>Kylin VPN</source>
-        <translation type="unfinished"></translation>
+        <translation>اسپاپVPN</translation>
     </message>
     <message>
         <location filename="../frontend/single-pages/singlepage.cpp" line="144"/>
         <source>kylin vpn applet desktop message</source>
-        <translation type="unfinished"></translation>
+        <translation>VPNتەڭشەگىن شىرە  بەتىندە كۅرسۅتۉۉ</translation>
     </message>
 </context>
 <context>
@@ -24,391 +24,391 @@
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="827"/>
         <source>MPPE encryption algorithm:</source>
-        <translation type="unfinished"></translation>
+        <translation>MPPE سىيفىرلوو ەسەپتۅۅ  ارعاسى:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="828"/>
         <source>Use Stateful encryption</source>
-        <translation type="unfinished"></translation>
+        <translation>ىشتەتىش  ابالى سىيفىرلاشقان</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="829"/>
         <source>Send PPP echo packets</source>
-        <translation type="unfinished"></translation>
+        <translation>pppقايتۇۇ كۅرسۅتۉۉ  بوقچوسۇنا جولدوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="830"/>
         <source>Authentication Mode:</source>
-        <translation type="unfinished"></translation>
+        <translation>دالىلدۅ كەبەتەسى ، تۇرپاتى :</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="831"/>
         <source>PAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>PAP دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="832"/>
         <source>CHAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>CHAP دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="833"/>
         <source>MSCHAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>MSCHAP دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="834"/>
         <source>MSCHAP2 authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>MSCHAP2 دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="835"/>
         <source>EAP authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>EAP دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="836"/>
         <source>Compression Mode:</source>
-        <translation type="unfinished"></translation>
+        <translation>پېرىسلاش ابالى:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="837"/>
         <source>Allow BSD data compression</source>
-        <translation type="unfinished"></translation>
+        <translation>BSD نى پېرىسلاشقا جول قويۇش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="838"/>
         <source>Allow Default data compression</source>
-        <translation type="unfinished"></translation>
+        <translation>Defaultنى پېرىسلاشقا جول قويۇش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="839"/>
         <source>Allow TCP header compression</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP نى پېرىسلاشقا جول قويۇش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="840"/>
         <source>Use protocol field compression negotiation</source>
-        <translation type="unfinished"></translation>
+        <translation>ىشتەتىش   كەلىشىمى چۅلكۅمۉندۅگۉ پېرىسلاش  كەلىشىمى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="841"/>
         <source>Use Address/Control compression</source>
-        <translation type="unfinished"></translation>
+        <translation>دايىن وردۇ/كونتىروللاپ پېرىسلاشنى ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="847"/>
         <source>All Available</source>
-        <translation type="unfinished"></translation>
+        <translation>باردىق بارلار</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="848"/>
         <source>128-bit</source>
-        <translation type="unfinished"></translation>
+        <translation>128 بىتتۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="849"/>
         <source>40-bit</source>
-        <translation type="unfinished"></translation>
+        <translation>40 بىتتۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1046"/>
         <source>Use custom gateway port</source>
-        <translation type="unfinished"></translation>
+        <translation>انىقتاما  بەرۉۉ تور  ۅتكۅلۉ ووزۇن  ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1047"/>
         <source>Use compression</source>
-        <translation type="unfinished"></translation>
+        <translation>پىرىستاپ ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1048"/>
         <source>Use a TCP connection</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP ۇلوو ىڭعايىن  قولدونۇش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1049"/>
         <source>Set virtual device type</source>
-        <translation type="unfinished"></translation>
+        <translation>جورۇما شايمان تىبىن بەلگىلۅۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1050"/>
         <source>Set virtual device name</source>
-        <translation type="unfinished"></translation>
+        <translation>جورۇما شايمان اتاعىن بەلگىلۅۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1051"/>
         <source>Limit TCP Maximum Segment Size(MSS)</source>
-        <translation type="unfinished"></translation>
+        <translation>TCP ەڭ چوڭ بۅلۉك چوڭدۇعۇن  چەكتۅۅ (MSS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1052"/>
         <source>Randomize remote hosts</source>
-        <translation type="unfinished"></translation>
+        <translation>ئىختىيارىيلاشقان ابالدا باش اپاراتتى ىرااقتان كونتىرول جاسوو ،اتقارۇۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1053"/>
         <source>IPv6 tun link</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv6 tunۇلوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1054"/>
         <source>Specify ping interval</source>
-        <translation type="unfinished"></translation>
+        <translation>pingدەۋرىيلىكىنى بەلگىلۅۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1055"/>
         <source>Specify exit or restart ping</source>
-        <translation type="unfinished"></translation>
+        <translation>جانىش ، قايتىش  كۅرۉنۉشتۅرۉ قايرا   قوزعولعون ping نى كۅرسۅتۉپ  بەرۉۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1056"/>
         <source>Specify max routes</source>
-        <translation type="unfinished"></translation>
+        <translation>چەكتۉۉ لىنىيانى كۅرسۅتۉپ  بەرۉۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1057"/>
         <source>Infinite retry on error</source>
-        <translation type="unfinished"></translation>
+        <translation>قاتاالىق  چىققاندا چەكسىز ابالدا قايرا قايتالوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1058"/>
         <source>Use custom key size</source>
-        <translation type="unfinished"></translation>
+        <translation>انىقتاما  بەرىلگەن سىرلۇۇ اچقىچ چوڭ كىچىكتىگىن  ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1059"/>
         <source>Choose</source>
-        <translation type="unfinished"></translation>
+        <translation>تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1062"/>
         <source>Use custom renegotiation interval</source>
-        <translation type="unfinished"></translation>
+        <translation>ۅز الدىنا قايرا سۆھبەتلىشىش ارالىعىن  ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1064"/>
         <source>Use custom tunnel Maximum Transmission Umit(MTU)</source>
-        <translation type="unfinished"></translation>
+        <translation>قاسىيەت تونەلىنىن ەڭ چوڭ جولدونمو ۇزاتىش  مۇئەسسەسەسى(MTU) نى ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1066"/>
         <source>Use custom UDP fragment size</source>
-        <translation type="unfinished"></translation>
+        <translation>انىقتاما  بەرىلگەن UDP مايدالاش  چوڭدۇعۇن  ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1068"/>
         <source>Accept authenticated packets from any address (Float)</source>
-        <translation type="unfinished"></translation>
+        <translation>ارقانداي دايىن وردۇ  قابىلدوو  جاسوو ،اتقارۇۇ  ( Float ) اتۇۇلدۇق  تەكشەرىش  ارقىلۇۇ ساندۇۇ بايانداما  بوقچوسۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1070"/>
         <source>Subject Match</source>
-        <translation type="unfinished"></translation>
+        <translation>باش سىيۇجەتتى  شايكەشتىرىپ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1071"/>
         <source>Key File</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉيۉندۉۉ ۅجۅت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1072"/>
         <source>Key Direction</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉيۉندۉۉ باعىت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1074"/>
         <source>Server Address</source>
-        <translation type="unfinished"></translation>
+        <translation>مۇلازىمەتىر دايىن وردۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1075"/>
         <source>Port</source>
-        <translation type="unfinished"></translation>
+        <translation>ووزۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1076"/>
         <source>Proxy USername</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكالەتەن ابونت ناامى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1077"/>
         <source>Proxy Password</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكالەتەن سىرلۇۇ نومۇر</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1079"/>
         <source>General</source>
-        <translation type="unfinished"></translation>
+        <translation>اداتتاعى ەرەجە</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1080"/>
         <source>TLS settings</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS تەڭشەكتەرى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1081"/>
         <source>Server Certificate Check</source>
-        <translation type="unfinished"></translation>
+        <translation>مۇلازىمەتىر كەنشكەسىن تەكشەرىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1083"/>
         <source>Use the previous authentication end (server) certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>مۇرۇنقۇ دالىلدۅ سوڭۇ ( مۇلازىمەتىر) كەنشكەسىن ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1085"/>
         <source>Verify peer (server) certificate nsCertType specification</source>
-        <translation type="unfinished"></translation>
+        <translation>تەڭ  تۉيۉن  ( مۇلازىمەتىر) كەنىشكەسى nsCertType نىڭ  كونكرېتلىقىنى دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1086"/>
         <source>Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>ۉلگۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1087"/>
         <source>Proxies</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكالەتەن</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1088"/>
         <source>Proxy Type</source>
-        <translation type="unfinished"></translation>
+        <translation>ۋاكالەتەن تۉرۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1089"/>
         <source>Security</source>
-        <translation type="unfinished"></translation>
+        <translation>قووپسۇزدۇق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1090"/>
         <source>HMAC Authentication</source>
-        <translation type="unfinished"></translation>
+        <translation>HMAC دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1100"/>
         <source>Input content</source>
-        <translation type="unfinished"></translation>
+        <translation>مازمۇن كىرگىزۉۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1102"/>
         <source>No</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1106"/>
         <source>Self-adaption</source>
-        <translation type="unfinished"></translation>
+        <translation>ۅز-ۅزىگە شايكەشۉۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1107"/>
         <source>Automatic</source>
-        <translation type="unfinished"></translation>
+        <translation>اپتوماتىك</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1112"/>
         <source>Exit</source>
-        <translation type="unfinished"></translation>
+        <translation>جانىش ، قايتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1113"/>
         <source>Restart</source>
-        <translation type="unfinished"></translation>
+        <translation>قايرا قوزعوتۇۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1115"/>
         <source>Don&apos;t verify certificate identification</source>
-        <translation type="unfinished"></translation>
+        <translation>كەنىشكە اتۇۇلدۇق  دالىلدۅۅنۉ ئىسپاتلىماڭ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1116"/>
         <source>Verify the entire subject exactly</source>
-        <translation type="unfinished"></translation>
+        <translation>پۉتۉن سىيۇجەتتى  بىر وتولۇ انىقتاش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1117"/>
         <source>Verify name exactly</source>
-        <translation type="unfinished"></translation>
+        <translation>اتاعىن انىقتاپ دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1118"/>
         <source>Verify name by prefix</source>
-        <translation type="unfinished"></translation>
+        <translation>اتاعىن الدى   قوشۇلما بويۇنچا  دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1120"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1123"/>
         <source>Server</source>
-        <translation type="unfinished"></translation>
+        <translation>مۇلازىمەتىر</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1121"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1124"/>
         <source>Client</source>
-        <translation type="unfinished"></translation>
+        <translation>ئەپ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1126"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1130"/>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1139"/>
         <source>None</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1127"/>
         <source>TLS-Certification</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS-دالىلدۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1128"/>
         <source>TLS-Encryption</source>
-        <translation type="unfinished"></translation>
+        <translation>TLS-سىيفىرلوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1134"/>
         <source>Not Required</source>
-        <translation type="unfinished"></translation>
+        <translation>تالاپ قىلىنبايت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1138"/>
         <source>Default</source>
-        <translation type="unfinished"></translation>
+        <translation>العاچىنداعى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1216"/>
         <source>Options:</source>
-        <translation type="unfinished"></translation>
+        <translation>تاندالباعاندار:</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1219"/>
         <source>Request an inner IP address</source>
-        <translation type="unfinished"></translation>
+        <translation>ىچكى  IP دايىن وردۇ ۅتۉنۉچ  جاسوو ،اتقارۇۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1220"/>
         <source>Enforce UDP encapsulation</source>
-        <translation type="unfinished"></translation>
+        <translation>UDP قاپتوونۇ اتقارماق  جاسوو ،اتقارۇۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1221"/>
         <source>Use IP compression</source>
-        <translation type="unfinished"></translation>
+        <translation>IP  پېرىسلاش ىشتەتىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1222"/>
         <source>Enable custom password suggestions</source>
-        <translation type="unfinished"></translation>
+        <translation>ۅزۉ  كۅرسۅتۉلگۅن  جاشىرۇۇن  نومۇر قوزعوتۇۇ سۇنۇشۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1382"/>
         <source>Choose a private key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكەلىك  اچقىچ تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnadvancedpage.cpp" line="1384"/>
         <source>Key Files (*.key *.pem *.der *.p12 *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉيۉندۉۉ  ۅجۅتتۅر (*.key *.pem *.der *.p12 *.pfx)</translation>
     </message>
 </context>
 <context>
@@ -416,96 +416,96 @@
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="331"/>
         <source>Type</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉرۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="332"/>
         <source>Name</source>
-        <translation type="unfinished"></translation>
+        <translation>ناامى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="333"/>
         <source>Static Key</source>
-        <translation type="unfinished"></translation>
+        <translation>تىنىچ  ابالىنداعى جاشىرۇۇن  اچقىچ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="334"/>
         <source>Local IP</source>
-        <translation type="unfinished"></translation>
+        <translation>جەردىك  دايىن وردۇIP</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="335"/>
         <source>Remote IP</source>
-        <translation type="unfinished"></translation>
+        <translation>ىرااقتان كونتىرول جاسايتۇرعان دايىن وردۇIP</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="336"/>
         <source>PIN Code</source>
-        <translation type="unfinished"></translation>
+        <translation>PIN قۇپۇيا نومۇرۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="337"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="849"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="856"/>
         <source>Password</source>
-        <translation type="unfinished"></translation>
+        <translation>جاشىرۇۇن  نومۇر</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="338"/>
         <source>NT Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>NT گۅۅلۅم</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="340"/>
         <source>Server Address</source>
-        <translation type="unfinished"></translation>
+        <translation>مۇلازىمەتىر دايىن وردۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="341"/>
         <source>Authentication Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>دالىلدۅ ۉلگۉسۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="342"/>
         <source>CA Certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>CA كەنەشكەسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="343"/>
         <source>User Certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>ىشتەتۉۉچۉ كەنىشكەسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="344"/>
         <source>Key Direction</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉيۉندۉۉ باعىت</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="345"/>
         <source>Private Key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكەلىك  اچقىچ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="346"/>
         <source>Private Key Password</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكەلىك  اچقىچ پارولۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="347"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="349"/>
         <source>Password Options</source>
-        <translation type="unfinished"></translation>
+        <translation>پارول  تاندالمالارى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="348"/>
         <source>Username</source>
-        <translation type="unfinished"></translation>
+        <translation>ىشتەتۉۉچۉ ناامى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="352"/>
         <source>Notice:
 If key direction is used, it must be opposite to the VPN side used. If &apos;1&apos; is used, the connection must use &apos;0&apos;. If you are not sure which value to use, please contact your system administrator.</source>
-        <translation type="unfinished"></translation>
+        <translation>دىققات:ەگەر  جاشىرۇۇن  اچقىچ  باعىتىن ىشتەتكەن  بولسو، ال  سۅزسۉز ىشتەتكەن  VPN ال چى مەنەن قاراما قارشى بولۇۇسۇ كەرەك. ەگەر  «1» نى ىشتەتكەن  بولسو، ال لاشتا سۅزسۉز «0» نى ىشتەتىش  كەرەك. ەگەر  قايسى نارقتۇۇنۇ  ىشتەتىش نى تۇراقتاشتىرا الباساڭىز ساامالىق  باشقارعۇۇچۇ مەنەن بايلانىش جاساڭ.</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="359"/>
@@ -513,104 +513,104 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="361"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="362"/>
         <source>Choose</source>
-        <translation type="unfinished"></translation>
+        <translation>تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="369"/>
         <source>None</source>
-        <translation type="unfinished"></translation>
+        <translation>جوق</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="373"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="379"/>
         <source>Save password only for this user</source>
-        <translation type="unfinished"></translation>
+        <translation>جالاڭ عانا  گەزەكتەكى  ابونت جاشىرۇۇن  نومۇر ساقتوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="374"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="380"/>
         <source>Save password for all users</source>
-        <translation type="unfinished"></translation>
+        <translation>باردىق ابونتتار ۉچۉن  جاشىرۇۇن  نومۇر ساقتوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="375"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="381"/>
         <source>Ask password every time</source>
-        <translation type="unfinished"></translation>
+        <translation>ار بىر ىرەت  جاشىرۇۇن  نومۇر سۇروو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="376"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="382"/>
         <source>Don&apos;t require a password</source>
-        <translation type="unfinished"></translation>
+        <translation>جاشىرۇۇن  نومۇر تالاپ قىلبوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="385"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="386"/>
         <source>Required</source>
-        <translation type="unfinished"></translation>
+        <translation>مۉلدۅ تولتۇرۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="854"/>
         <source>Certificate(TLS)</source>
-        <translation type="unfinished"></translation>
+        <translation>كەنەشكە(TLS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="855"/>
         <source>Static key</source>
-        <translation type="unfinished"></translation>
+        <translation>تىنىچ  ابالىنداعى اچقىچ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="857"/>
         <source>Password and certificate(TLS)</source>
-        <translation type="unfinished"></translation>
+        <translation>جاشىرۇۇن  نومۇر جانا كەنىشكە(TLS)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="866"/>
         <source>Certificate/Private key</source>
-        <translation type="unfinished"></translation>
+        <translation>كەنىشكە/جەكە  اچقىچ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="867"/>
         <source>Certificate/ssh-agent</source>
-        <translation type="unfinished"></translation>
+        <translation>كەنەشكە/ ssh-agent</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="868"/>
         <source>Smart card</source>
-        <translation type="unfinished"></translation>
+        <translation>روحىي جۅندۅمدۉلۉك كارتوچكا</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="923"/>
         <source>Choose a private key</source>
-        <translation type="unfinished"></translation>
+        <translation>جەكەلىك  اچقىچ تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="925"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="964"/>
         <source>Key Files (*.key *.pem *.der *.p12 *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>تۉيۉندۉۉ  ۅجۅتتۅر (*.key *.pem *.der *.p12 *.pfx)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="936"/>
         <source>Choose a CA certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>CA كەنىشكەسى تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="938"/>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="951"/>
         <source>CA Files (*.pem *.der *.p12 *.crt *.cer *.pfx)</source>
-        <translation type="unfinished"></translation>
+        <translation>CA  ۅجۅتتۅرۉ (*.pem *.der *.p12 *.crt *.cer *.pfx)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="949"/>
         <source>Choose a User certificate</source>
-        <translation type="unfinished"></translation>
+        <translation>ىشتەتۉۉچۉ كەنىشكەسى تانداش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnconfigpage.cpp" line="962"/>
         <source>Choose a Static key</source>
-        <translation type="unfinished"></translation>
+        <translation>تىنىچ  ابالىنداعى اچقىچ تانداش</translation>
     </message>
 </context>
 <context>
@@ -635,22 +635,22 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="172"/>
         <source>Advanced</source>
-        <translation type="unfinished"></translation>
+        <translation>جوعورۇ چەك</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="177"/>
         <source>Auto Connection</source>
-        <translation type="unfinished"></translation>
+        <translation>اپتوماتتىك ۇلوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="178"/>
         <source>Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>ارعادان  قالتىرىش</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpndetail.cpp" line="179"/>
         <source>Confirm</source>
-        <translation type="unfinished"></translation>
+        <translation>بەكىتۉۉ</translation>
     </message>
 </context>
 <context>
@@ -658,47 +658,47 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="170"/>
         <source>IPv4 Config</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv4 تەڭشەگى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="171"/>
         <source>Address</source>
-        <translation type="unfinished"></translation>
+        <translation>دايىن وردۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="172"/>
         <source>Netmask</source>
-        <translation type="unfinished"></translation>
+        <translation>بۅلۉمچۅ  تور  جاشىرىن نومۇرۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="173"/>
         <source>Default Gateway</source>
-        <translation type="unfinished"></translation>
+        <translation>العاچىنداعى تور  ۅتكۅلۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="174"/>
         <source>DNS Server</source>
-        <translation type="unfinished"></translation>
+        <translation>DNS  سان قاينارى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="175"/>
         <source>Search Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>تور  اتاعىن ىزدۅۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="176"/>
         <source>DHCP Client ID</source>
-        <translation type="unfinished"></translation>
+        <translation>DHCP ئەپ IDسى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="178"/>
         <source>Auto(DHCP)</source>
-        <translation type="unfinished"></translation>
+        <translation>اپتوماتتىك(DHCP)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv4page.cpp" line="179"/>
         <source>Manual</source>
-        <translation type="unfinished"></translation>
+        <translation>قول  ارقىلۇۇ اراكەتدەندۇرۉۉ</translation>
     </message>
 </context>
 <context>
@@ -706,42 +706,42 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="159"/>
         <source>IPv6 Config</source>
-        <translation type="unfinished"></translation>
+        <translation>IPv6 تەڭشەگى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="160"/>
         <source>Address</source>
-        <translation type="unfinished"></translation>
+        <translation>دايىن وردۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="161"/>
         <source>Netmask</source>
-        <translation type="unfinished"></translation>
+        <translation>بۅلۉمچۅ  تور  جاشىرىن نومۇرۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="162"/>
         <source>Default Gateway</source>
-        <translation type="unfinished"></translation>
+        <translation>العاچىنداعى تور  ۅتكۅلۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="163"/>
         <source>DNS Server</source>
-        <translation type="unfinished"></translation>
+        <translation>DNS  سان قاينارى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="164"/>
         <source>Search Domain</source>
-        <translation type="unfinished"></translation>
+        <translation>تور  اتاعىن ىزدۅۅ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="166"/>
         <source>Auto(DHCP)</source>
-        <translation type="unfinished"></translation>
+        <translation>اپتوماتتىك(DHCP)</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnipv6page.cpp" line="167"/>
         <source>Manual</source>
-        <translation type="unfinished"></translation>
+        <translation>قول  ارقىلۇۇ اراكەتدەندۇرۉۉ</translation>
     </message>
 </context>
 <context>
@@ -749,19 +749,19 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="61"/>
         <source>Not connected</source>
-        <translation type="unfinished"></translation>
+        <translation>ۇلانباعان</translation>
     </message>
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="146"/>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="161"/>
         <source>Disconnect</source>
-        <translation type="unfinished"></translation>
+        <translation>ۉزۉلۉش</translation>
     </message>
     <message>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="148"/>
         <location filename="../frontend/list-items/vpnlistitem.cpp" line="159"/>
         <source>Connect</source>
-        <translation type="unfinished"></translation>
+        <translation>ۇلانۇۇ</translation>
     </message>
 </context>
 <context>
@@ -774,7 +774,7 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/single-pages/vpnpage.cpp" line="264"/>
         <source>VPN Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN تەڭشەكتەرى</translation>
     </message>
 </context>
 <context>
@@ -782,43 +782,43 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="58"/>
         <source>create VPN</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN قۇرۇۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="92"/>
         <source>VPN Type</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN تۉرۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="120"/>
         <source>VPN Name</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN ناامى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="126"/>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="147"/>
         <source>Required</source>
-        <translation type="unfinished"></translation>
+        <translation>مۉلدۅ تولتۇرۇ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="142"/>
         <source>VPN Server</source>
-        <translation type="unfinished"></translation>
+        <translation>VPN  سان قاينارى</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="168"/>
         <source>Auto Connection</source>
-        <translation type="unfinished"></translation>
+        <translation>اپتوماتتىك ۇلوو</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="170"/>
         <source>Confirm</source>
-        <translation type="unfinished"></translation>
+        <translation>بەكىتۉۉ</translation>
     </message>
     <message>
         <location filename="../frontend/vpndetails/vpnaddpage.cpp" line="172"/>
         <source>Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>ارعادان  قالتىرىش</translation>
     </message>
 </context>
 <context>
@@ -833,7 +833,7 @@ If key direction is used, it must be opposite to the VPN side used. If &apos;1&a
     <message>
         <location filename="../frontend/vpnobject.cpp" line="50"/>
         <source>vpn tool</source>
-        <translation type="unfinished"></translation>
+        <translation>vpn قۇرالى</translation>
     </message>
 </context>
 </TS>
